test/e2e/operator.go

```diff
@@ -272,3 +276,108 @@ var _ = Describe("ARO Operator - Conditions", func() {
 		Expect(err).NotTo(HaveOccurred())
 	})
 })
+
+var _ = Describe("ARO Operator - Azure NSG Reconciler", func() {
+	var vnetName, location, resourceGroup, workerSubnet, masterSubnet string
+	var dummynsg mgmtnetwork.SecurityGroup
+	ctx := context.Background()
+	enableReconcileNSG := func() {
+		instance, err := clients.AROClusters.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		if !instance.Spec.Features.ReconcileNSGs {
```

Comment:
Please add TODO and create an gh issue to remove this once it is enabled by default

Reply:
https://github.com/Azure/ARO-RP/issues/1735